app/mailers/team_mailer.rb

```diff
@@ -0,0 +1,31 @@
+# An ApplicationMailer subclass that is responsible for informing users about team-related changes via email
+class TeamMailer < ApplicationMailer
+  after_action :prevent_delivery_to_unsubscribed_users
+
+  # Notifies a user that he/she was added to a team
+  def user_added_to_team(user, team)
+    @user = user
+    @team = team
```

Comment:
Is there programmatically a reason for creating and assigning instance variables with values of method parameters here? I have seen this a couple of time in this rails project already... Is it because you need `@user` in `prevent_delivery_to_unsubscribed_users` or again a codeclimate issue?

Reply:
Or maybe just a rails convention?

Reply:
It's needed in the templates. But looks ugly, yes

Reply:
@lheilmann Yep – sadly there's no way around this because of Rails' template-rendering technique. Actually, CodeClimate's *Reek* does not like these instance variables as well:

![2018-01-28 23 54 34](https://user-images.githubusercontent.com/807665/35488309-b445c936-0486-11e8-8046-5d418c5e33f3.jpg)

Since those CodeClimate smells have occurred several times in this project now (both in non-solvable contexts like this one and in solvable contexts where the apparent code smell really is a code smell), I told @birneamstiel as your current SM delegate to talk with the SM team about configuring Reek to work better with Rails-based Ruby projects as documented in its official Readme: https://github.com/troessner/reek/blob/master/README.md#working-with-rails

Reply:
Alright! Great idea :+1: